File: PianoKeyboard_java_eclipse/src/com/teuskim/pianokeyboard/BlackKeyView.java
package com.teuskim.pianokeyboard;

import android.content.Context;
import android.util.AttributeSet;

public class BlackKeyView extends KeyView {

	public BlackKeyView(Context context, AttributeSet attrs, int defStyle) {
		super(context, attrs, defStyle);
	}

	public BlackKeyView(Context context, AttributeSet attrs) {
		super(context, attrs);
	}

	public BlackKeyView(Context context) {
		super(context);
	}

	@Override
	protected int getLayoutResId() {
		return R.layout.key_black;
	}

	@Override
	protected int getNormalImageResId() {
		return R.drawable.key_black;
	}

	@Override
	protected int getPressedImageResId() {
		return R.drawable.key_black_pressed;
	}

	@Override
	public int getKeyType() {
		return KEY_TYPE_BLACK;
	}

}

File: PianoKeyboard_kotlin/app/src/main/java/com/teuskim/pianokeyboard/BlackKeyView.kt
package com.teuskim.pianokeyboard

import android.content.Context
import android.util.AttributeSet

class BlackKeyView : KeyView {

    protected override val layoutResId: Int
        get() = R.layout.key_black

    protected override val normalImageResId: Int
        get() = R.drawable.key_black

    protected override val pressedImageResId: Int
        get() = R.drawable.key_black_pressed

    override val keyType: Int
        get() = KeyView.KEY_TYPE_BLACK

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context) : super(context) {}

}
